daffodil-io/src/main/scala/org/apache/daffodil/processors/charset/BitsCharset.scala

```diff
@@ -55,7 +55,7 @@ trait BitsCharset extends Serializable {
   def aliases: Seq[String] = Nil
   def bitWidthOfACodeUnit: Int // in units of bits
   def requiredBitOrder: BitOrder
-  def mandatoryBitAlignment: Int
+  def mandatoryBitAlignment: Int // ignored when dfdlx:alignmentViaSkip is "yes"
```

Comment:
I think we need a test to see what happens if we're using regular 8-bit ascii charset encoding, but it isn't 8-bit aligned because we forced it to not do alignment via dfdlx:alignmentViaSkip. Similarly for utf-16 that isn't byte aligned.

Getting a runtime error here, so long as the diagnostic indicates the character is not aligned properly, is the right behavior we want.

Reply:
I agree, let's add a test to determine whether the diagnostic looks appropriate.